packages/react/src/QuestionnaireForm/QuestionnaireFormItem/QuestionnaireFormItem.tsx

```diff
@@ -429,3 +441,48 @@ function typedValueToString(typedValue: TypedValue | undefined): string | undefi
   }
   return typedValue.value.toString();
 }
+
+function getItemsByLinkId(responses: QuestionnaireResponseItem[], linkId: string): QuestionnaireResponseItem[] {
+  let result: QuestionnaireResponseItem[] = [];
+
+  for (const item of responses) {
+    // If the linkId matches, add it to the result array
+    if (item.linkId === linkId) {
+      result.push(item);
+    }
+
+    // If the current item has nested items, search them too
+    if (item.item) {
+      result = result.concat(getItemsByLinkId(item.item, linkId));
+    }
+  }
+  return result;
+}
+
+function getItemValue(answer: QuestionnaireResponseItemAnswer): any {
+  const itemValue = getTypedPropertyValue(
+    { type: 'QuestionnaireItemAnswerOption', value: answer },
+    'value'
+  ) as TypedValue;
+  if (itemValue?.type === 'Coding') {
```

Comment:
I'm trying to remember our previous conversation about this... Is there a reason we can't handle this special case in the form item rather than here?

If nothing else, I'm nervous that this will not be sufficient.  Many `Coding` objects only have a `code`, and no `display`.